gcpp/programs/mptidy.1
.TH mptidy 1 "August 2013" "cmassoc-tools-1.9.1" "Motley Tools"

.SH NAME
mptidy - Linux man page tidy program

.SH SYNOPSIS
.BR mptidy
.RI [ options ]
.IR file
.RI [ file ] 
.RI [ ... ]

.PP
.BR mptidy
.RI [ options ]
.RI [< file ]
.RI [> file ]

.SH DESCRIPTION

.PP
Update file header with configuration strings and format file body such that each statement and sentence starts on a new line.
The header is the first line and starts with the command ".TH".
The remainder of thath line consists of the program name, man section and three title strings.
The file basename becomes the program name.
The file extension becomes the man section and should be one digit [0-9].
The three titles are read from a named profile, the default profile or internal program constants.
The profile section can be specified.

.SH OPTIONS
Long options are not supported.
All options are single letters.

.TP
.RB - o
Print a template profile on stdout.
The template uses default program values.
Output can be directed to a file, edited and read back whenever the program is next executed.
This option takes precedence over normal formatting operations.

.TP
-\fBp\fI profile\fR
The profile name.
If this option is present then the program attempts to open and read the profile.
The program will report an error and terminated if the named profile file does not exist.
If this option is omitted the program attempts to read the default profile.
The program does not report an error when the default profile is missing.
The default profile is "\fB/etc/mp.ini\fR".

.TP
-\fBs\fI section\fR
The profile section name.
If this option is present the the program will search the named profile or default profile for the named section.
If this option is omitted the progam will search for the default section, instead.
The default section is named "\fBdefault\fR".

.TP
.RB - ? ,-- help
Print program help summary on stdout.
This option takes precedence over previous command line options.

.TP
.RB - ! ,-- version
Print program version information on stdout.
This option takes precedence over previous command line options.

.SH ARGUMENTS

.TP
.IR file
The name of man page.
This file is replaced and a numbered backup is created.
Use program \fBrex\fR to recover a prior version.

.SH PROFILE
The default profile looks like:

.PP
 [default]
 project=Motley Tools
 package=cmassoc-tools-1.5.9
 release=Feb 2004

.TP
.BR project
The project name appears at the top center of the page.
A reasonable value is the company or project name.
The default is "Motley Tools".

.TP
.BR package
The package name appears in the lower left of the page.
A reasonable value is the archive name with revision numbers.
The default is "cmassoc-tools-x.x.x".

.TP
.BR release
The release data appears at the bottom center of the page.
A reasonable value is the current month and year.
The default is the current month and year.

.SH EXAMPLES
The following example formats man page \fBabc.7\fR using default program values unless configuration provile \fB/etc/mp.ini\fR is present.
In that event, values will be read from there.

.PP
 # mptidy abc.7

.PP
The following example output a template configuration profile section.

.PP
 # mptidy -o
 [default]
 project=Motley Tools
 package=mtools-1.8.1
 program=program
 release=data/time

.SH SEE ALSO
.BR afm ( 7 ),
.BR cfm ( 7 ),
.BR ctidy ( 7 ),
.BR csstidy ( 7 ),
.BR dnstidy ( 7 ),
.BR htmltidy ( 7 ),
.BR xmltidy ( 7 ),
.BR rex ( 7 )

.SH CREDITS
 Charles Maier <cmaier@cmassoc.net>



gcpp/programs/mptidy.cpp
/*====================================================================*
 *
 *   mptidy.cpp - man page tidy program;
 *
 *.  Motley Tools by Charles Maier <cmaier@cmassoc.net>;
 *:  Copyright 2010-2013 by Charles Maier Associates;
 *;  Licensed under the Internet Software Consortium License
 *
 *--------------------------------------------------------------------*/

/*====================================================================*
 *   system header files;
 *--------------------------------------------------------------------*/

#include <cstdlib>
#include <iostream>

/*====================================================================*
 *   custom header files;
 *--------------------------------------------------------------------*/

#include "../classes/ogetoptv.hpp"
#include "../classes/opathspec.hpp"
#include "../classes/ofileopen.hpp"
#include "../classes/oprofile.hpp"
#include "../classes/omptidy.hpp"

/*====================================================================*
 *   custom source files;
 *--------------------------------------------------------------------*/

#ifndef MAKEFILE
#include "../classes/ogetoptv.cpp"
#include "../classes/oputoptv.cpp"
#include "../classes/oversion.cpp"
#include "../classes/oerror.cpp"
#include "../classes/oflagword.cpp"
#include "../classes/ofileopen.cpp"
#include "../classes/ofilespec.cpp"
#include "../classes/opathspec.cpp"
#include "../classes/oprofile.cpp"
#include "../classes/ocontext.cpp"
#include "../classes/owildcard.cpp"
#include "../classes/oescape.cpp"
#include "../classes/oascii.cpp"
#include "../classes/osource.cpp"
#include "../classes/oindent.cpp"
#include "../classes/oswitch.cpp"
#include "../classes/otext.cpp"
#include "../classes/omptidy.cpp"
#endif

#define PROFILE_NAME "/etc/mp.ini"     
#define SECTION_NAME "default"     

#define MPTIDY_S_PROJECT "Motley Tools"     
#define MPTIDY_S_PACKAGE "mtools-1.9.0"     
#define MPTIDY_S_PROGRAM "mtools.7"     
#define MPTIDY_S_RELEASE __DATE__

/*====================================================================*
 *
 *   int main (int argc, char const * argv []);
 *
 *
 *.  Motley Tools by Charles Maier <cmaier@cmassoc.net>;
 *:  Copyright 2001-2006 by Charles Maier Associates;
 *;  Licensed under the Internet Software Consortium License
 *
 *--------------------------------------------------------------------*/

int main (int argc, char const * argv [])

{
	static char const * optv [] =
	{
		"mop:s:",
		oPUTOPTV_S_FILTER,
		"format C/C++ source code with preamble annotations",
		"m\tprint example manpage on stdout",
		"o\tprint default profile on stdout",
		"p s\tprofile is (s) [" LITERAL (PROFILE_NAME) "]",
		"s s\tsection is (s) [" LITERAL (SECTION_NAME) "]",
		(char const *) (0)
	};
	ogetoptv getopt;
	ofileopen fileopen;
	opathspec pathspec;
	oprofile config;
	omptidy object;
	char const * profile = PROFILE_NAME;
	char const * section = SECTION_NAME;
	signed (omptidy::* method) (signed) = & omptidy::tidy;
	signed c;
	while (~ (c = getopt.getoptv (argc, argv, optv)))
	{
		switch (c)
		{
		case 'm':
			object.example ();
			std::exit (0);
		case 'o':
			config.write (SECTION_NAME);
			config.write (oMPTIDY_S_PROJECT, MPTIDY_S_PROJECT);
			config.write (oMPTIDY_S_PACKAGE, MPTIDY_S_PACKAGE);
			config.write (oMPTIDY_S_PROGRAM, MPTIDY_S_PROGRAM);
			config.write (oMPTIDY_S_RELEASE, MPTIDY_S_RELEASE);
			std::exit (0);
		case 'p':
			profile = getopt.optarg ();
			break;
		case 's':
			section = getopt.optarg ();
			break;
		default: 
			break;
		}
	}
	object.project (config.string (profile, section, oMPTIDY_S_PROJECT, MPTIDY_S_PROJECT));
	object.package (config.string (profile, section, oMPTIDY_S_PACKAGE, MPTIDY_S_PACKAGE));
	object.program (config.string (profile, section, oMPTIDY_S_PROGRAM, MPTIDY_S_PROGRAM));
	object.release (config.string (profile, section, oMPTIDY_S_RELEASE, MPTIDY_S_RELEASE));
	if (! getopt.argc ())
	{
		(object.* method) (std::cin.get ());
	}
	while (getopt.argc () && * getopt.argv ())
	{
		if (fileopen.openedit (* getopt.argv ()))
		{
			object.filename (* getopt.argv ());
			(object.* method) (std::cin.get ());
			fileopen.close ();
		}
		getopt++;
	}
	std::exit (0);
}

